### PR TITLE
CORE-11709: Point HSMStore to use CryptoRepository

### PR DESCRIPTION
**Pull request by fowlerrr**
Pointing those two classes to use the CryptoRepository means less code duplication and a path towards eventually removing them.  There's a lot of tests that need to be managed, however, in order to get that to work so this is a simpler and more expedient way to achieve that result.

Also ensuring that uses of CryptoRepository are closed to save database connections.

### CODE REVIEW

File: components/crypto/crypto-softhsm-impl/src/test/kotlin/net/corda/crypto/softhsm/impl/SigningKeyStoreUnitTest.kt
Comment (fowlerrr) [inline components/crypto/crypto-softhsm-impl/src/test/kotlin/net/corda/crypto/softhsm/impl/SigningKeyStoreUnitTest.kt:1]: I need more time looking at these tests but I think they're already covered so I might just deleted the file rather than make them fit what we have now

File: components/crypto/crypto-persistence-impl/build.gradle [hunk 1]
```diff
@@ -24,6 +24,7 @@ dependencies {
     implementation project(":components:crypto:crypto-component-core-impl")
     implementation project(":components:crypto:crypto-persistence-model")
     implementation project(":components:crypto:crypto-persistence")
+    implementation project(":components:crypto:crypto-softhsm-impl")
```
Comment (dickon) [inline components/crypto/crypto-persistence-impl/build.gradle:27]: surprised that's not a cycle

File: components/crypto/crypto-softhsm-impl/src/test/kotlin/net/corda/crypto/softhsm/impl/V1SigningKeyStoreTest.kt [hunk 2]
```diff
@@ -208,4 +207,6 @@ class V1SigningKeyStoreTest {
         assertThat(tenantCap.allValues.single()).isEqualTo("tenant")
         assertThat(fullIdsCap.allValues.single()).isEqualTo(listOf(hashB.toString()))
     }
+
```
Comment (dickon) [inline components/crypto/crypto-softhsm-impl/src/test/kotlin/net/corda/crypto/softhsm/impl/V1SigningKeyStoreTest.kt:210]: unnecessary whitespace

(not a merge blocker)

**Review by dickon — APPROVED** [review]
LGTM, nice
